Replace Ansible with Shell script

ngosang opened the pull request:
* Remove Ansible and all Python packages
* Reduce image size by 456 MB (689 MB => 233 MB uncompressed)
* Fixes some open issues, for example, root password is not required if the database already exists.
* Show install and startup traces (traces and errors were hidden by Ansible)

Resolves #303 #269 #270 #256

ngosang commented:
@wallabag/core I did many tests but this change is big. It keeps all the functionality and the environment variables are the same.

ngosang commented:
Ready to review. If you give me permissions in this repository I can take triage the open issues. Half of them are already fixed with the latest changes.

j0k3r commented on Dockerfile:
```diff
@@ -54,18 +56,20 @@ RUN set -ex \
  && curl -s https://getcomposer.org/installer | php \
  && mv composer.phar /usr/local/bin/composer \
- && composer selfupdate 2.2.12
+ && composer selfupdate 2.2.12 \
```
```suggestion
 && composer selfupdate 2.2.18 \
```

ngosang replied:
fixed

j0k3r reviewed (commented):
You need to rebase against master to fix conflict (I guess it's about the composer version).
Maybe @Kdecherf can have a look before merging?

Kdecherf commented:
@ngosang @j0k3r fyi I need to take time to review and test these changes, but I won't be able to do it before two weeks

ngosang commented:
I can wait, but I hope that at some point it can be merged. It has cost me a lot of effort. If it could be merged before the next release of Wallabag that would be perfect.

Kdecherf commented on Dockerfile:
```diff
@@ -54,18 +56,20 @@ RUN set -ex \
  && ln -sf /dev/stderr /var/log/nginx/error.log \
  && curl -s https://getcomposer.org/installer | php \
  && mv composer.phar /usr/local/bin/composer \
- && composer selfupdate 2.2.18
+ && composer selfupdate 2.2.18 \
```
composer could be installed through an intermediate Docker container, for example:

```
COPY --from=composer:2.2.18 /usr/bin/composer /usr/local/bin/composer
```

ngosang replied:
I tried and the image size is the same. The Docker image build process is a bit slower with your solution because the Composer Docker image has many layers. You can open another PR if you want.

Kdecherf replied:
Noted

Kdecherf commented on root/entrypoint.sh:
```diff
@@ -1,27 +1,105 @@
+                mysql -h "${SYMFONY__ENV__DATABASE_HOST}" --port "${SYMFONY__ENV__DATABASE_PORT}" -uroot -p"${MYSQL_ROOT_PASSWORD}" \
+                    -e "GRANT ALL PRIVILEGES ON ${SYMFONY__ENV__DATABASE_NAME}.* TO '${SYMFONY__ENV__DATABASE_USER}'@'%';"
+            fi
+            install_wallabag
+        else
+            echo "WARN: MySQL database is already confiugred. Remove the environment variable with root password."
```
typo on "configured"

ngosang replied:
fixd

Kdecherf commented on root/entrypoint.sh:
```diff
@@ -1,27 +1,105 @@
+            fi
+            install_wallabag
+        else
+            echo "WARN: Postgres database is already confiugred. Remove the environment variable with root password."
```
typo on configured

ngosang replied:
fixed

Kdecherf commented on root/etc/wallabag/parameters.template.yml:
```diff
@@ -0,0 +1,63 @@
+    database_socket: null
+    database_charset: ${SYMFONY__ENV__DATABASE_CHARSET:-utf8}
+
+    domain_name: ${SYMFONY__ENV__DOMAIN_NAME:-https://your-wallabag-url-instance.com}
```
We should replace the default url by something like `itworks.wallabag.org`, see https://github.com/wallabag/wallabag/issues/5890

ngosang replied:
I didn't change the behavior in this PR, Ansible install set the same value.
I think the best solution will be to remove the default value so this Environment Variable is mandatory. This could beak some installations so it has to be documented. Please, do the change in other PR or open an issue to keep the track.

Kdecherf replied:
Agree'd

Kdecherf commented on root/etc/wallabag/parameters.template.yml:
```diff
@@ -0,0 +1,63 @@
+    locale: ${SYMFONY__ENV__LOCALE:-en}
+
+    # A secret key that's used to generate certain security-related tokens
+    secret: ${SYMFONY__ENV__SECRET:-ovmpmAWXRCabNlMgzlzFXDYmCFfzGv}
```
We should remove the default secret here

j0k3r replied:
Can't we generate a random one each the file is _built_?

ngosang replied:
I didn't change the behavior in this PR, Ansible install set the same secret. It's easy to generate random strings in Bash but it will be tricky to detect if it's the first run or not.
I think the best solution will be to remove the default value so this Environment Variable is mandatory. This could beak some installations so it has to be documented.

Kdecherf replied:
Noted

Kdecherf commented on root/entrypoint.sh:
```diff
@@ -1,27 +1,105 @@
+
+provisioner() {
+    SYMFONY__ENV__DATABASE_DRIVER=${SYMFONY__ENV__DATABASE_DRIVER:-pdo_sqlite}
+    POPULATE_DATABASE=${POPULATE_DATABASE:-True}
```
I don't know if we want to have POPULATE_DATABASE set to true by default.

On a side-note I would be for replacing T with t as it is a Python thing. (`True` => `true` in conditions below)

ngosang replied:
I didn't change the behavior in this PR, Ansible install set the same value. I think this en var is not well documented. Maybe we could improve this in other PR but for now I don't want to break things for the users. Or can open an issue to keep the track...

Kdecherf replied:
Noted

Kdecherf reviewed (commented):
Awesome work @ngosang!

I've quickly tested the PR against SQLite, MariaDB and PostgreSQL with 3 scenarios:
- First boot, uninstalled
- Reboot with root db credentials still there
- Reboot without root db credentials

Looks good, I made some comments though

Kdecherf commented:
Side note: there may be some cleanup to do on the commit log (rewording and/or squashing some commits)

ngosang commented:
@Kdecherf @j0k3r I fixed the typos and squashed all commits into one. I agree with the other suggestions but I think they are out of the scope of this PR. This PR replaces Ansible with bash trying to have the same default values for environment variables.  Also this PR is already tested. Remaining changes could be implemented in other PRs.

Kdecherf reviewed (approved):
LGTM

Kdecherf commented:
@ngosang you're right, we'll handle those improvements in separate PR.

Thanks for your work
